src/main/java/net/mcreator/ui/modgui/ModElementChangedListener.java

```diff
@@ -115,9 +112,14 @@ private boolean isGenericComponent(Component component) {
 	}
 
 	@Override default void actionPerformed(ActionEvent e) {
-		if (e.getSource() instanceof JComboBox) {
-			if (e.getModifiers() != 0)
+		// if this is a JComboBox, we check if any modifier keys or mouse buttons are currently used
+		if (e.getSource() instanceof JComboBox box && e.getModifiers() != 0) {

```

Comment:
I don't understand the filtering on modifiers, especially not on default JComboBox. Java developers have thought this system and when action listener is triggered, it is for a reason. I really can't see a reason to mess with this system

Reply:
> if this is a JComboBox, we check if any modifier keys or mouse buttons are currently used

This comment also just describes what we can read one line below, and is thus not useful. Your comment should explain reasoning, not something we can read one line below...

Reply:
`// <...> we check if the change comes from user interaction with UI`

Reply:
User interraction is with or without modifiers.

Your comment does not explain to me why this check is here

There should be clear comment explaining this..

Reply:
Reading source code, mouse buttons are treated as modifiers too.

Reply:
> User interraction is with or without modifiers.

> Your comment does not explain to me why this check is here

Reply:
> mouse buttons are treated as modifiers too.

When I do regular click, modifiers are 0, but I am sure I am using mouse button to click on the dropdown

Reply:
Tested by adding 

![image](https://github.com/MCreator/MCreator/assets/16374228/04cbe1ad-a76c-477c-87db-d91aa7345d40)

Reply:
This matches `BUTTON1_MASK`.

Reply:
I get 16 when no user action happens...

Reply:
In other words, you can't rely in getModifiers to determine if action source is "user" or "automation"

Reply:
`ActionListener` can be replaced with `ItemListener`, it is said to trigger on selection changes only.